addons/pos_loyalty/static/src/overrides/models/pos_store.js

```diff
@@ -374,45 +647,48 @@ patch(PosStore.prototype, {
      * IMPROVEMENT: It would be better to update the local order object instead of creating a new one.
      *   - This way, we don't need to remember the lines linked to negative coupon ids and relink them after pushing the order.
      */
-    async push_single_order(...args) {
-        const [orderToPush] = args;
-
-        // Keep some linked records to relink them after pushing the order to the server.
-        const lineCouponMap = orderToPush.lines.reduce((agg, line) => {
-            if (line.coupon_id && line.coupon_id.id < 0) {
-                return { ...agg, [line.uuid]: line.coupon_id.id };
-            } else {
-                return agg;
-            }
-        }, {});
-        const lineRewardProductMap = orderToPush.lines.reduce((agg, line) => {
-            if (line.reward_product_id) {
-                return { ...agg, [line.uuid]: line.reward_product_id.id };
-            } else {
-                return agg;
-            }
-        }, {});
+    preSyncAllOrders(orders) {
+        super.preSyncAllOrders(orders);
 
-        const [updatedOrder] = await super.push_single_order(...args);
+        for (const order of orders) {
+            this.couponByLineUuidCache = order.lines.reduce((agg, line) => {
```

Comment:
This should be a mistake because `couponByLineUuidCache` and `rewardProductByLineUuidCache` are shared between multiple orders and their value are based on the last order.